website_field_autocomplete/static/src/js/field_autocomplete.js

```diff
@@ -0,0 +1,69 @@
+/* Copyright 2016 LasLabs Inc.
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+ */
+
+odoo.define("website_field_autocomplete.field_autocomplete", function (require) {
+    "use strict";
+
+    var snippet_animation = require("website.content.snippets.animation");
+
+    snippet_animation.registry.field_autocomplete = snippet_animation.Animation.extend({
+        selector: ".js_website_autocomplete",
+
+        /* Query remote server for autocomplete suggestions
+         * @param request object Request from jQueryUI Autocomplete
+         * @param response function Callback for response, accepts array of str
+         */
+        autocomplete: function (request, response) {
+            var self = this;
+            var domain = [[this.queryField, "ilike", request.term]];
+            if (this.add_domain) {
+                domain = domain.concat(this.add_domain);
+            }
+            return $.ajax({
+                dataType: "json",
+                url: "/website/field_autocomplete/" + self.model,
+                method: "GET",
+                data: {
+                    domain: JSON.stringify(domain),
+                    fields: JSON.stringify(self.fields),
+                    limit: self.limit,
+                },
+            }).then(function (records) {
+                var data = records.reduce(function (a, b) {
+                    a.push({label: b[self.displayField], value: b[self.valueField]});
+                    return a;
+                }, []);
+                response(data);
+                return records;
+            });
+        },
+
+        /* Return arguments that are used to initialize autocomplete */
+        autocompleteArgs: function () {
+            var self = this;
```

Comment:
Use 'arrow functions' to avoid 'self' usage.

Reply:
@Tardo I tried to change as little as possible the original code of the module, I modified it with your suggestions please check when you have time, thanks.